src/GameLoop.tsx

```diff
@@ -534,14 +532,37 @@ export function GameLoop() {
     }
 
     return (
-        <BoardStyled>
-            <PlayerComponent
-                isVisible={isPlayerVisible}
-                adjustedCells={playerPhysics.adjustedCells}
-            />
-            <BoardComponent
-                boardCellMatrix={boardPhysics.boardCellMatrix}
-            />
-        </BoardStyled>
+        <>
+            <div style={AppStyle}>
+                <BoardStyled>
+                    <PlayerComponent
+                        isVisible={isPlayerVisible}
+                        adjustedCells={playerPhysics.adjustedCells}
+                    />
+                    <BoardComponent
+                        boardCellMatrix={boardPhysics.boardCellMatrix}
+                    />
+                </BoardStyled>
+                <WordList displayedWords={matchedWords} />
+            </div>
+        </>
     );
 }
+
+const wordStyle = {
+    display: "block",
+    background: "yellow",
+};
+
+const WordList = React.memo(({ displayedWords }) => {
+    return (
+        <div display={"flex"} flex-direction={"column"}>
+            <div>Matched Words ({totalMatchedWords})</div>
+            <>
+                {displayedWords.map((word) => (
+                    <div key={word} style={wordStyle}>{word}</div>
```

Comment:
I don't think `word` is a good key here because it's not unique.  I'd use the index, with the first word matched being `0`.  Even if the old words are not shown anymore, keep using the original indices (though I think you should show them all), that way the index key is unique and unchanging.

Reply:
Fixed